server/src/main/scala/drt/server/feeds/lgw/LGWFeed.scala

```diff
@@ -0,0 +1,277 @@
+          BaggageReclaimId = Try(n \\ "FIDSBagggeHallActive" text).getOrElse(""),
+          FlightID = 0,
+          AirportID = "LGW",
+          Terminal = mappedTerminal,
+          rawICAO = (n \\ "AirlineICAO" text) + parseFlightNumber(n),
+          rawIATA = (n \\ "AirlineIATA" text) + parseFlightNumber(n),
+          Origin = parseOrigin(n),
+          SchDT = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text ).equals("SCT") ).map(n=>n text).getOrElse(""),
+          Scheduled = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text ).equals("SCT") ).map(n=> services.SDate.parseString(n text).millisSinceEpoch).getOrElse(0),
+          PcpTime = 0,
+          LastKnownPax = None)
+        log.info(s"parsed arrival: $arrival")
+        arrival
+      }
+
+      result.toList
+    }
+
+    val ourSendReceive: HttpRequest => Future[HttpResponse] = sendReceive
+
+    val resultPipeline: pipelining.WithTransformerConcatenation[HttpRequest, Future[List[Arrival]]] = (
+      addHeader("Authorization", wrapHeader)
+        ~> ourSendReceive
+        ~> toArrivals
+      )
+
+    resultPipeline(Delete(serviceBusUri))
+  }
+
+  private def parseFlightNumber(n: Node) = {
+    ((n \ "FlightLeg").head \ "LegIdentifier").head \ "FlightNumber" text
+  }
+
+  def parseStatus(n: Node): String = {
+    ((n \ "FlightLeg").head \ "LegData").head \ "OperationalStatus" text
+  }
+
+  def parseOrigin(n: Node): String = {
+    ((n \ "FlightLeg").head \ "LegIdentifier").head \ "DepartureAirport" text
+  }
+
+  def parseRunwayId(n: Node): Option[String] = {
+    (n \\ "AirportResources" \ "Resource").find(n => (n \ "@DepartureOrArrival" text).equals("Arrival")).map(n => n \\ "Runway" text)
+  }
+
+  def parsePaxCount(n: Node, qualifier: String): Option[Int] = {
+    (n \\ "PaxCount").find(n => (n \ "@Qualifier" text).equals(qualifier) && (n \ "@Class").isEmpty).map(n => (n text).toInt)
+  }
+
+  def parseDateTime(n: Node, operationQualifier: String, timeType: String): Option[String] = {
+    (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text ).equals("SCT") ).map(n=>n text)
+    (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals(operationQualifier) && (n \ "@TimeType" text).equals(timeType)).map(n => n text)
+  }
+
+  def createAzureSamlAssertionAsString(privateKey: Array[Byte], certificate: Array[Byte]): String = {
+    val assertion = createAzureSamlAssertion(privateKey, certificate)
+
+    // welcome to java and its horrendous mutating method magic. the following two lines
+    // do something important to the signature
+    Configuration.getMarshallerFactory.getMarshaller(assertion).marshall(assertion)
```

Comment:
This could probably be put into a method that describes what's happening, although I appreciate you probably didn't write this code.

Reply:
I'll google it to see what it's doing, I'm not sure myself. Lance's comment looks accurate and made me laugh.

Reply:
I love a good joke as much as the next person, but sarcasm isn't always that helpful for somebody coming to figure out WTF is happening later. 

Feel free to leave it though, if it's working and doesn't seem that consequential how it is doing it.